### PR TITLE
Always use url as reference key in cc.loader

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#4896

Changes proposed in this pull request:
 * Always use url as reference key in cc.loader
 * Fix lazy loading asset couldn’t be auto released issue

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/load-pipeline/uuid-loader.js [hunk 1]
```diff
@@ -58,13 +61,13 @@ function loadDepends (pipeline, item, asset, tdInfo, deferredLoadRawAssetsInRunt
                 // skip preloading raw assets
                 var url = info.url;
                 obj[prop] = url;
+                dependKeys.push(url);
```
Comment (jareguo) [inline cocos2d/core/load-pipeline/uuid-loader.js:64]: 这里不应该 push 吧？否则就不能跳过资源加载了。

**Review by jareguo — CHANGES_REQUESTED** [review]

File: cocos2d/core/load-pipeline/uuid-loader.js (hunk 1 above)
Comment (pandamicro) [reply cocos2d/core/load-pipeline/uuid-loader.js:64]: 为啥不能跳过？dependKeys 只是用来记录引用关系，跟加载没关系啊，加载是 depends 列表。如果这里不 push 的话，auto release 的时候会找不到 raw asset 的引用，然后就不会释放

File: cocos2d/core/platform/CCAssetLibrary.js [hunk 2]
```diff
@@ -219,6 +225,7 @@ var AssetLibrary = {
             }
             else {
                 if (asset.constructor === cc.SceneAsset) {
+                    // We know scene is a raw asset, so we can pass uuid to this API directly
```
Comment (jareguo) [inline cocos2d/core/platform/CCAssetLibrary.js:228]: scene 不是 raw asset 吧？

**Review by jareguo — CHANGES_REQUESTED** [review]

File: cocos2d/core/load-pipeline/uuid-loader.js (hunk 1 above)
Comment (jareguo) [reply cocos2d/core/load-pipeline/uuid-loader.js:64]: 哦，看错了，看成了 depends

**Review by jareguo — APPROVED** [review]

File: cocos2d/core/platform/CCAssetLibrary.js (hunk 2 above)
Comment (pandamicro) [reply cocos2d/core/platform/CCAssetLibrary.js:228]: 写错了，是 is not a raw asset

**Pull request merged**